### PR TITLE
feat: SfProductCard with SfColorPicker on category page

### LINKED ISSUE #950: [FEATURE]Product listing- add button '+ Colors' on mobile
**Describe how this feature will help us improve**
Easy and quick way of choosing color of product

**Describe the solution you'd considered**
Add button '+ Colors' on mobile on product listing as below:
+Color

After pressing the '+ Colors' button a list of available colors of the product appears as below:
+Color1
After pressing color, the choice is remembered. Then if user adds the product to the cart, the product is added with chosen color.

**Acceptance criteria**

- [ ]   the '+ Colors' button is added to the product listing on mobile
- [ ]  after pressing the '+ Color' a list of available colors appears
- [ ] chosen color is remembered as long as the user is on that product listing
- [ ] pressing any spot other than color square closes the list of available colors
- [ ] after choosing the color, when user adds the product to the cart, it is added with tat chosen color

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes #950 

# Scope of work
Colors added to SfProductCard

# Screenshots of visual changes

Zrzut ekranu z 2020-10-30 15-11-17

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Comment by aniamusial** [comment]
@justyna-13 can you please resolve the conflicts? :)

File: packages/vue/src/components/pages/category/Category.vue [hunk 1]
```diff
@@ -515,6 +517,28 @@ export default {
...
+              color: "#F59F93",
+              selected: false,
+            },
+            {
+              label: "Citrus",
+              value: "citrus",
+              color: "#FFEE97",
+              selected: false,
+            },
+          ],
```
Comment (AdamPawlinski) [inline packages/vue/src/components/pages/category/Category.vue:541]: What do you think about adding a separate file for this data and importing them, like `internalData.js` file in templates?

**Review by AdamPawlinski — COMMENTED** [review]
Works nicely, except there is no possibility to close color picker on mobile. Probably because there is this condition in ProductCard: 
```
:is-open="!isMobile || openColorPicker"
```
I suppose it should close on click and maybe should have close button added, or do I think wrong?

File: packages/vue/src/components/pages/category/Category.vue (hunk 1 above)
Comment (justyna-13) [reply packages/vue/src/components/pages/category/Category.vue:541]: I can add a separate file but I'm wondering if we want to show it in source code :) what do you think?

Comment (AdamPawlinski) [reply packages/vue/src/components/pages/category/Category.vue:541]: You are right, let's stay with this :)

**Comment by RafalDobrowolski** [comment]
I've tested. It works  when I go to mobile view

Screenshot 2022-02-18 at 13 03 11

But when I use mouse to resize window without switching to mobile view, it doesn't work. I think it's related with `mobile-observer`.

In my opinion when customer open page on mobile device everything should work, because mobile view should be detected.

Screenshot 2022-02-18 at 13 08 03

File: packages/vue/src/components/pages/home/Home.vue [hunk 2]
```diff
@@ -397,6 +567,15 @@ export default {
       return (this.products[index].isInWishlist =
         !this.products[index].isInWishlist);
     },
+    handleSelectedColor(color, index) {
+      this.products[index].colors.map((el) => {
+        if (el.label === color.label) {
+          el.selected = !el.selected;
+        } else {
+          el.selected = false;
+        }
```
Comment (RafalDobrowolski) [inline packages/vue/src/components/pages/home/Home.vue:576]:
```suggestion
     el.selected = el.label === color.label ? !el.selected : false;
```

File: packages/vue/src/components/pages/category/Category.vue [hunk 3]
```diff
@@ -695,6 +873,15 @@ export default {
     toggleWishlist(index) {
       this.products[index].isInWishlist = !this.products[index].isInWishlist;
     },
+    handleSelectedColor(color, index) {
+      this.products[index].colors.map((el) => {
+        if (el.label === color.label) {
+          el.selected = !el.selected;
+        } else {
+          el.selected = false;
```
Comment (RafalDobrowolski) [inline packages/vue/src/components/pages/category/Category.vue:881]:
```suggestion
      el.selected = el.label === color.label ? !el.selected : false;
```

File: packages/vue/src/components/pages/home/Home.vue (hunk 2 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/pages/home/Home.vue:576]: I made this change

File: packages/vue/src/components/pages/category/Category.vue (hunk 3 above)
Comment (AdamPawlinski) [reply packages/vue/src/components/pages/category/Category.vue:881]: I made this change too.

**Comment by AdamPawlinski** [comment]
I think that it's finally well done. @justyna-13 you can apply this change that Rafał suggest and for me it's okey :+1:

**Review by AdamPawlinski — APPROVED** [review]

**Comment by justyna-13** [comment]
@RafalDobrowolski done :)

**Review by RafalDobrowolski — APPROVED** [review]

**Pull request merged**